src/neo/SmartContract/ApplicationEngine.cs

```diff
@@ -115,6 +115,10 @@ public partial class ApplicationEngine : ExecutionEngine
         /// </summary>
         public IReadOnlyList<NotifyEventArgs> Notifications => notifications ?? (IReadOnlyList<NotifyEventArgs>)Array.Empty<NotifyEventArgs>();
 
+        private static ulong persistingNonce;
+        private static ulong nextNonce;
```

Comment:
Why static?

Reply:
There might be multiple `ApplicationEngin` instances while executing the `persistingBlock`, but the `nextNonce` should be consistent with the `persistingBlock`.

Reply:
But it can be modified by other instances. I think we should remove `static`.

Reply:
> But it can be modified by other instances. I think we should remove `static`.

that is exactly what we want, we only need one nextNonce instance for each Block. Otherwise different transactions well have  same nonce value.

Reply:
It has been fixed.